Lib/typing.py

```diff
@@ -3203,7 +3202,7 @@ def reveal_type(obj: T, /) -> T:
     return obj
 
 
-def dataclass_transform(
+def dataclass_transform[T](
```

Comment:
This feels slightly strange to me, as (to me) it implies that `dataclass_transform` is a function generic around the type variable `T`. But that's not really true; it's a function that returns an identity function. I skimmed PEP 695, and it doesn't seem to mention how such cases should be handled.

Reply:
IIRC we did think about this and decided that this is indeed how such cases should be spelled (since `Callable` doesn't have a way yet to spell "generic").

Reply:
I think we shouldn't do this since it confuses scoping and much of the point of PEP 695 is to clarify scoping. We can use a callback protocol to spell this.

I'd brought this up somewhere before PEP 695 was accepted, but didn't get much of a response. Presumably since it's not the most common thing and I think e.g. pyre might not even support `-> Callable[[T], T]`

Reply:
Changed to a callback protocol